pyanaconda/modules/security/certificates/certificates_interface.py

```diff
@@ -0,0 +1,83 @@
+#
+# DBus interface for the certificate module.
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from dasbus.server.interface import dbus_interface
+from dasbus.server.property import emits_properties_changed
+from dasbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.modules.common.base import KickstartModuleInterfaceTemplate
+from pyanaconda.modules.common.constants.objects import CERTIFICATES
+from pyanaconda.modules.common.structures.security import CertificateData
+from pyanaconda.modules.common.containers import TaskContainer
+
+
+@dbus_interface(CERTIFICATES.interface_name)
+class CertificatesInterface(KickstartModuleInterfaceTemplate):
+    """DBus interface for the certificate installation module."""
+
+    def connect_signals(self):
+        super().connect_signals()
+        self.watch_property("Certificates", self.implementation.certificates_changed)
+
+    @property
+    def Certificates(self) -> List[Structure]:
+        """All certificates.
+
+        :return: a list of certificate DBus Structures
+        """
+        return CertificateData.to_structure_list(self.implementation.certificates)
+
+    @Certificates.setter
+    @emits_properties_changed
+    def Certificates(self, certificates: List[Structure]):
```

Comment:
Yes, I agree.
Initially I added it mostly for testing :).
I didn't think it would be fundamentally more insecure. Maybe anyone can use ReadKickstart(string) API on the security module instead to do the same thing, but yes there seem to be no valid reason to provide setter API.
I'll try to make it read-only property.

Reply:
Done